GenderPayGap.WebUI/Controllers/Admin/AdminDataMigrationController.cs

```diff
@@ -0,0 +1,648 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+using System.Threading;
+using Autofac;
+using GenderPayGap.Core;
+using GenderPayGap.Core.Interfaces;
+using GenderPayGap.Database;
+using GenderPayGap.Database.Models;
+using GenderPayGap.WebUI.Helpers;
+using GenderPayGap.WebUI.Models.Admin;
+using GovUkDesignSystem.Parsers;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using Newtonsoft.Json;
+
+namespace GenderPayGap.WebUI.Controllers
+{
+    [Authorize(Roles = LoginRoles.GpgAdmin)]
+    [Route("admin")]
+    public class AdminDataMigrationController : Controller
+    {
+
+        private readonly IDataRepository dataRepository;
+
+        public AdminDataMigrationController(IDataRepository dataRepository)
+        {
+            this.dataRepository = dataRepository;
+        }
+
+
+        [HttpGet("data-migration/export-all")]
+        public IActionResult ExportAllData()
+        {
+            if (dataRepository.GetAll<User>().Count() > 0)
+            {
+                return Unauthorized();
+            }
+
+            var viewModel = new AdminFileUploadViewModel();
+            return View("DataMigrationFromFile", viewModel);
+        }
+
+        [HttpGet("data-migration/export-all-as-file-download")]
+        public IActionResult ExportAllDataAsFileDownload()
+        {
+            string allDataString = LoadAllDataFromDatabaseInfoJsonString();
+            byte[] allDataBytes = Zip(allDataString);
+
+            var fileContentResult = new FileContentResult(allDataBytes, "application/json")
+            {
+                FileDownloadName = "AllData.json.gz"
+            };
+
+            return fileContentResult;
+        }
+
+        [AllowAnonymous]
+        [HttpGet("data-migration/export-all")]

```

Comment:
I'm a bit confused about having two HttpGet with the same url

Reply:
Oops, one of these methods was added as a merge error - I'll remove it.
Thanks for spotting this!